lib/internal/test_runner/test.js

```diff
@@ -408,18 +408,24 @@ class Test extends AsyncResource {
     }
   }
 
-  matchesTestNamePatterns() {
-    const matchesByNameOrParent = ArrayPrototypeSome(testNamePatterns, (re) =>
-      RegExpPrototypeExec(re, this.name) !== null,
-    ) ||
-    this.parent?.matchesTestNamePatterns();
+  testMatchesPatterns() {
```

Comment:
I think we could reuse the existing function. Instead of trying to combine the `testNamePatterns` and `testSkipPatterns` logic in a single function, you could pass one set of patterns to the existing function. Then, update the calling logic in the `Test` constructor as needed.